### PR TITLE
#169258604 a user should be able to book an accommodation facility

### PR DESCRIPTION
**Pull request by hezronkimutai**
#### What does this PR do?
This PR adds a feature that will enable users to book accommodation facilities
#### Description of Task to be completed?
- Design the UI as per the mock up
- Connect the UI the the redux store
- Add actions that will perform asynchronous actions functions
#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend`
- `git checkout ft-book-accommodation-169258604`
- To run the test, do `npm run test`
- To start the application, run `npm run dev`
#### Any background context you want to provide?
 Before running the application ensure you have a `.env` file which is up to date, Use `.env.example` file as a guide.
#### What are the relevant pivotal tracker stories?
[#169258604](https://www.pivotaltracker.com/n/projects/2407428/stories/169258604)
#### Screenshots (if appropriate)
Screen Shot 2020-02-06 at 11 59 34
Screen Shot 2020-02-06 at 12 01 47

Screen Shot 2020-02-06 at 12 00 00

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -96,6 +97,7 @@ const App = () => (
               <UserRolesView />) : (<Redirect to="/Dashboard" />))}
           />
           <Route path="/accommodations" exact component={ViewAccomodation} />
+          <Route path="/bookings" exact component={bookAccommodation} />
```
Comment (NiyongaboEric) [inline src/App.js]: I think `/bookings/create` route is better. Then we shall `/bookings` routes to display all bookings.

File: src/containers/accommodation/bookAccommodation.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  render() {
+    const { display, popUpAction } = this.props;
+    return (
+      <Dashboard>
+        <PopUp
+          popUp={(
+            <BookAccommodation
+              id={23}
+              commentId={3}
+            />
```
Comment (NiyongaboEric) [inline src/containers/accommodation/bookAccommodation.js]: Are you hard coding the data?

File: src/redux/actions/bookings/bookAnAccommodation.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    const bookAccommodationSuccess = await apiCall
+      .post(API_URL, data, { headers: HEADERS_REQUEST });
+    notificationSuccess(bookAccommodationSuccess.data.message);
+    dispatch({
+      type: BOOK_ACCOMMODATION_SUCCESS,
+      payload: bookAccommodationSuccess.data,
+    });
+  } catch (error) {
+    notificationSuccess('Ooops! you have provided the wrong details');
+    dispatch({
```
Comment (NiyongaboEric) [inline src/redux/actions/bookings/bookAnAccommodation.js]: I  think it is notificationError

**Review by NiyongaboEric — CHANGES_REQUESTED** [review]
Good job @hezronkimutai. I would like to request to have the same booking folder structure.

**Review by nziokaivy — CHANGES_REQUESTED** [review]
Nice work @hezronkimutai! Just one tiny change you could make.
Once a user has booked you can redirect them to view their booking made

**Review by NiyongaboEric — APPROVED** [review]

**Review by victkarangwa — COMMENTED** [review]
Good job @hezronkimutai 
Just remember to keep us in the loop by updating the screenshots on PR description.

**Review by shemaeric — APPROVED** [review]

**Review by victkarangwa — APPROVED** [review]

**Review by nziokaivy — APPROVED** [review]

**Pull request merged**